src/sensor/ping.cpp

```diff
@@ -359,25 +359,33 @@ void Ping::flash(const QString& fileUrl, bool sendPingGotoBootloader, int baud,
 
     qCDebug(PING_PROTOCOL_PING) << "Finish connection.";
     // TODO: Move thread delay to something more.. correct.
-    QThread::msleep(1000);
-    link()->finishConnection();
+    auto flashSensor = std::function<void(void)>();
+    auto finishConnection = std::function<void(void)>();
+
+    flashSensor = [=] {
+        flasher()->setBaudRate(baud);
+        flasher()->setFirmwarePath(fileUrl);
+        flasher()->setLink(link()->configuration()[0]);
+        flasher()->setVerify(verify);
+        flasher()->flash();
+    };
+
+    finishConnection = [=] {
+        QThread::msleep(1000);
```

Comment:
should this line be here? it looks like we are still sleeping the main thread.

Reply:
lost code, will be removed.